internal/services/synapse/synapse_workspace_sql_aad_admin_resource.go

```diff
@@ -0,0 +1,146 @@
+package synapse
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/synapse/mgmt/2021-03-01/synapse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/synapse/parse"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/services/synapse/validate"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/validation"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/timeouts"
+	"github.com/hashicorp/terraform-provider-azurerm/utils"
+)
+
+func resourceSynapseWorkspaceSqlAADAdmin() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceSynapseWorkspaceSqlAADAdminCreateUpdate,
+		Read:   resourceSynapseWorkspaceSqlAADAdminRead,
+		Update: resourceSynapseWorkspaceSqlAADAdminCreateUpdate,
+		Delete: resourceSynapseWorkspaceSqlAADAdminDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.WorkspaceSqlAADAdminID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"synapse_workspace_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ValidateFunc: validate.WorkspaceID,
+			},
+
+			"login": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+			},
```

Comment:
can revalidate this? I imagine empty or a single letter or symbols are now allowed here